frontend/src/utilities/useWatchIntegrationComponents.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+import { useAppSelector } from '~/redux/hooks';
+import { OdhApplication } from '~/types';
+import { getIntegrationAppEnablementStatus } from '~/services/integrationAppService';
+import { allSettledPromises } from '~/utilities/allSettledPromises';
+import { POLL_INTERVAL } from './const';
+import { isIntegrationApp } from './utils';
+
+type OdhIntegrationApplication = OdhApplication & { spec: { internalRoute: string } };
+
+export const useWatchIntegrationComponents = (
+  components?: OdhApplication[],
+): { checkedComponents: OdhApplication[]; isIntegrationComponentsChecked: boolean } => {
+  const [isIntegrationComponentsChecked, setIsIntegrationComponentsChecked] = React.useState(false);
+  const forceUpdate = useAppSelector((state) => state.forceComponentsUpdate);
+  const initForce = React.useRef<number>(forceUpdate);
+  const integrationComponents = React.useMemo(
+    () =>
+      components?.filter((component): component is OdhIntegrationApplication =>
+        isIntegrationApp(component),
+      ),
+    [components],
+  );
+  const [newComponents, setNewComponents] = React.useState<OdhApplication[]>([]);
+
+  const updateComponentEnablementStatus = async (
+    integrationComponentList: OdhIntegrationApplication[],
+    componentList: OdhApplication[],
+  ): Promise<void> => {
+    const updatePromises = integrationComponentList.map(async (component) => {
+      const response = await getIntegrationAppEnablementStatus(component.spec.internalRoute).catch(
+        (e) => ({
+          isInstalled: false,
+          isEnabled: false,
+          canInstall: false,
+          error: e.message ?? e.error, // might be an error from the server, might be an error in the network call itself
+        }),
```

Comment:
```suggestion
        } satisfies IntegrationAppStatus),
```
This will help keep the type consistent if we change it -- sorry for missing that during the debug session

Reply:
Updated